### PR TITLE
Fix to accept non-partial optimizers as well in OTXLitModule

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Case:
```
model = OTXResNet50(num_classes=num_classes)
optimizer = optim.SGD(model.parameters(), 
                        lr=0.01, 
                        weight_decay=1e-4, 
                        momentum=0.9)

T_max = 100000
scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max, eta_min=0)
lit_module = OTXLitModule(otx_model=model, optimizer=optimizer, scheduler=scheduler, torch_compile=False)
```

Need to add Test Cases as well

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/model/module/base.py [hunk 1]
```diff
@@ -103,9 +104,15 @@ def configure_optimizers(self) -> dict[str, Any]:
 
         :return: A dict containing the configured optimizers and learning-rate schedulers to be used for training.
         """
-        optimizer = self.hparams.optimizer(params=self.parameters())
+        if isinstance(self.hparams.optimizer, partial):
+            optimizer = self.hparams.optimizer(params=self.parameters())
+        else:
+            optimizer = self.hparams.optimizer
```
Comment (vinnamkim) [inline src/otx/core/model/module/base.py]: Please change constructor's argument typing as well.
```suggestion
        optimizer = self.hparams.optimizer(params=self.parameters()) if isinstance(self.hparams.optimizer, OptimizerCallable) else self.hparams.optimizer
```
Change `scheduler` as well.

Comment (harimkang) [reply src/otx/core/model/module/base.py]: https://github.com/openvinotoolkit/training_extensions/pull/2840/commits/3c24df17dccc6c610712a02b1d089cd409990837 - Thanks, I also fixed it to `if callable` instead of `if isinstance`.

**Review by sungmanc — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]
